CMakeLists.txt

```
@@ -26,6 +26,10 @@ set (SOURCES
 
 set(STRING_ENCODING_TYPE "ICONV" CACHE STRING "Set the way strings have to be encoded (ICONV|NONE|...)")
 
+if (WIN32)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
```

Is it necessary to wrap this setting into the `if (WIN32)` condition? I usually see `CMakeLists.txt` without it (https://github.com/robotology/how-to-export-cpp-library/blob/afb21efb655e7b1cc11636c3d42aad9e02fb626f/CMakeLists.txt#L48), but not sure if it doesn't fail under non-Windows systems in that case.

---

It is not necessary and it doesn't fail under Linux for instance. I've added the `if` to make it explicit for those who read `CMakeLists.txt` but I fixed it in the suggestion as you proposed.

```suggestion
set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```